src/push_service/cdn.rs

```diff
@@ -417,18 +417,25 @@ impl PushService {
 
         trace!("digested content");
 
-        self.request(
+        let mut request = self.request(
             Method::PATCH,
             Endpoint::cdn(3, resumable_url.path()),
             HttpAuthOverride::Unidentified,
-        )?
-        .header("Tus-Resumable", "1.0.0")
-        .header("Upload-Offset", resume_info.content_start)
-        .header("Upload-Length", buf.len())
-        .header(CONTENT_TYPE, content_type)
-        .body(buf)
-        .send()
-        .await?;
+        )?;
+
+        for (key, value) in headers {
+            request = request.header(key, value);
+        }
+
+        request
+            .header("Tus-Resumable", "1.0.0")
+            .header("Upload-Offset", resume_info.content_start)
+            .header("Upload-Length", buf.len())
+            .header(CONTENT_TYPE, content_type)
```

Comment:
It's probably ok as-is but the other clients hard-code `content-type`, eg https://github.com/signalapp/Signal-iOS/blob/main/SignalServiceKit/Upload/UploadEndpointCDN3.swift#L87 . The [TUS](https://tus.io/protocols/resumable-upload#patch) protocol apparently mandates it as such for PATCH requests.

Reply:
Very good point. I really wonder why we even have this parameters in the first place 🤔